gcloud/storage/_helpers.py

```diff
@@ -79,10 +79,12 @@ def _patch_property(self, name, value):
     def _set_properties(self, value):
         """Set the properties for the current object.
 
-        :type value: dict
+        :type value: dict or :class:`gcloud.storage.batch._FutureDict`
         :param value: The properties to be set.
         """
         self._properties = value
+        if hasattr(value, 'owner'):
+            value.owner = self
```

Comment:
Yes, but then when the batch completes, how will `Batch.finish` assign to those future objects when we do have a real dict? It has to have a reference to those objects (which it won't, since `api_request` is not responsible for creating objects) and it needs to know the order that the objects came in, to correspond to the batch response.